pkg/ntpmonitor/ntpmonitor.go

```diff
@@ -151,9 +152,12 @@ func (n *NTPMonitor) Start() {
 
 	delta := time.Duration(n.cfg.MaxTimeDelta) * time.Second
 	log.Logger.Info("ntp monitoring starting")
+
+	//nolint:gosec
+	r := rand.New(rand.NewSource(time.Now().UTC().UnixNano())) // initialize local pseudorandom generator //nolint:gosec
```

Comment:
One downside to this is the source is not thread safe, unlike the default rand source. Unfortunately golang doesn't export this source. It's not that big of a deal though since this isn't used in such a way, but it'd be nice to get thread safety in case there's a change in the future.

What do you think about making the random generator optional in `RandomChoice`? Callers that don't pass in `rand.Rand` will call `rand.Intn`, otherwise it'll call `r.Intn`.

Reply:
Going to resolve so we can unblock PR